x-pack/test/api_integration/apis/ml/data_frame_analytics/delete.ts

```diff
@@ -0,0 +1,237 @@
+          .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+          .set(COMMON_HEADERS)
+          .expect(404);
+
+        expect(body.error).to.eql('Not Found');
+        expect(body.message).to.eql('Not Found');
+      });
+
+      describe('with deleteTargetIndex setting', function () {
+        const analyticsId = `${jobId}_2`;
+        const destinationIndex = generateDestinationIndex(analyticsId);
+
+        before(async () => {
+          await es.indices.create({ index: destinationIndex });
+          await ml.api.assertIndicesExist(generateDestinationIndex(analyticsId));
+        });
+
+        after(async () => {
+          await es.indices.delete({ index: destinationIndex, ignore: [404] });
+        });
+
+        it('should delete job and target index by id', async () => {
+          const { body } = await supertest
+            .delete(`/api/ml/data_frame/analytics/${analyticsId}`)
+            .query({ deleteTargetIndex: true })
+            .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+            .set(COMMON_HEADERS)
+            .expect(200);
+
+          expect(body.analyticsJobDeleted.success).to.eql(true);
+          expect(body.targetIndexDeleted.success).to.eql(true);
+          expect(body.targetIndexPatternDeleted.success).to.eql(false);
+          expect(jobDeleted(analyticsId)).to.eql(true);
+
+          const targetIndexExists = await es.indices.exists({
+            index: destinationIndex,
+            allowNoIndices: false,
+          });
+          expect(targetIndexExists).to.eql(false);
+        });
+      });
+      describe('with deleteTargetIndex setting', function () {
+        const analyticsId = `${jobId}_3`;
+        const destinationIndex = generateDestinationIndex(analyticsId);
+
+        before(async () => {
+          // Mimic real job by creating index pattern after job is created
+          await ml.testResources.createIndexPatternIfNeeded(destinationIndex);
+        });
+
+        after(async () => {
+          await ml.testResources.deleteIndexPattern(destinationIndex);
+        });
+
+        it('should delete job and index pattern by id', async () => {
+          const { body } = await supertest
+            .delete(`/api/ml/data_frame/analytics/${analyticsId}`)
+            .query({ deleteIndexPattern: true })
+            .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+            .set(COMMON_HEADERS)
+            .expect(200);
+          expect(body.analyticsJobDeleted.success).to.eql(true);
+          expect(body.targetIndexDeleted.success).to.eql(false);
+          expect(body.targetIndexPatternDeleted.success).to.eql(true);
+          expect(jobDeleted(analyticsId)).to.eql(true);
+
+          // Check if index pattern was deleted
+          const indexPatternId = await ml.testResources.getIndexPatternId(analyticsId);
+          expect(indexPatternId).to.eql(undefined);
```

Comment:
I'd rather have this as a service method in `ml.testResources` with some retry to make it more stable. Something like `waitForIndexPatternNotToExist`, similar to [`waitForAnomalyDetectionJobNotToExist`](https://github.com/elastic/kibana/blob/master/x-pack/test/functional/services/ml/api.ts#L293) .